<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PackagingHelperDialog</class>
 <widget class="QDialog" name="PackagingHelperDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Packaging Helper</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../res.qrc">
    <normaloff>:/res/icons/package_wrench.png</normaloff>:/res/icons/package_wrench.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="welcomePage">
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="welcomeLabel">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;Welcome to the Packaging Helper tool.&lt;/span&gt;&lt;/p&gt;&lt;p&gt;As the name suggests, this tool is going to try to help you with packaging your unfulfilled orders. It will give you a list of orders that you can quickly browse and filter. Only packaging-relevant information will be displayed.&lt;/p&gt;&lt;p&gt;For Shipping, select all the methods you want to filter on.&lt;br/&gt;&lt;br/&gt;For Products, the logic is hopefully mostly self-explanatory. If you select 3 products as &amp;quot;Must include&amp;quot;, only orders with all 3 products will be accepted by the filter, if you select 3 products as &amp;quot;May include&amp;quot;, any order that has at least one of these products will be accepted by the filter. If you select &amp;quot;Must NOT include&amp;quot;, any order that contains that product will be excluded, even if it contains products with May or Must logic.&lt;br/&gt;&lt;br/&gt;After that click &amp;quot;Start&amp;quot; to get started.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;This is a work in progress, if you have ideas on what to improve and/or add, hop on into Discord and let us know!&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="welcomeLine">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,2">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QLabel" name="shippingSelectionLabel">
             <property name="text">
              <string>Shipping:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTreeWidget" name="shippingSelectionTree">
             <property name="indentation">
              <number>0</number>
             </property>
             <attribute name="headerVisible">
              <bool>false</bool>
             </attribute>
             <column>
              <property name="text">
               <string notr="true">1</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="productSelectionLabel">
             <property name="text">
              <string>Products:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTreeWidget" name="productSelectionTree">
             <property name="styleSheet">
              <string notr="true">QTreeWidget::item { padding: 1px; }</string>
             </property>
             <attribute name="headerMinimumSectionSize">
              <number>20</number>
             </attribute>
             <attribute name="headerDefaultSectionSize">
              <number>50</number>
             </attribute>
             <attribute name="headerStretchLastSection">
              <bool>false</bool>
             </attribute>
             <column>
              <property name="text">
               <string notr="true">Product</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Logic</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Min Qty</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Max Qty</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="matchingOrdersLabel">
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>% orders match the selected filters</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="packagingPage">
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QSplitter" name="packagingSplitter">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="childrenCollapsible">
          <bool>false</bool>
         </property>
         <widget class="QWidget" name="layoutWidget">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QTreeWidget" name="orderListTree">
             <property name="showDropIndicator" stdset="0">
              <bool>true</bool>
             </property>
             <property name="indentation">
              <number>0</number>
             </property>
             <property name="headerHidden">
              <bool>true</bool>
             </property>
             <column>
              <property name="text">
               <string notr="true">1</string>
              </property>
             </column>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="orderListSortlabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Sort by:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="orderListSortComboBox">
               <item>
                <property name="text">
                 <string>Order Id</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Item count</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Weight</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Shipping method</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Total</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="layoutWidget_2">
          <layout class="QFormLayout" name="formLayout">
           <item row="1" column="0">
            <widget class="QLabel" name="orderNameLabel">
             <property name="text">
              <string>Name:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="orderNameValueLabel">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="orderCountryLabel">
             <property name="text">
              <string>Country:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="orderCountryValueLabel">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="orderWeightLabel">
             <property name="text">
              <string>Weight:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="orderWeightValueLabel">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="orderTrackingLabel">
             <property name="text">
              <string>Tracking:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLabel" name="orderTrackingValueLabel">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="orderShippingLabel">
             <property name="text">
              <string>Shipping method:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QLabel" name="orderShippingValueLabel">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="orderTotalLabel">
             <property name="text">
              <string>Total:</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QLabel" name="orderTotalValueLabel">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="customerNoteLabel">
             <property name="text">
              <string>Customer note:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QPlainTextEdit" name="customerNoteTextEdit">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>42</height>
              </size>
             </property>
             <property name="verticalScrollBarPolicy">
              <enum>Qt::ScrollBarAsNeeded</enum>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="plainText">
              <string/>
             </property>
            </widget>
           </item>
           <item row="8" column="0" colspan="2">
            <widget class="QLabel" name="orderItemsLabel">
             <property name="text">
              <string>Items in this order:</string>
             </property>
            </widget>
           </item>
           <item row="9" column="0" colspan="2">
            <widget class="QTreeWidget" name="orderItemTree">
             <property name="alternatingRowColors">
              <bool>false</bool>
             </property>
             <property name="indentation">
              <number>0</number>
             </property>
             <attribute name="headerVisible">
              <bool>false</bool>
             </attribute>
             <column>
              <property name="text">
               <string/>
              </property>
             </column>
            </widget>
           </item>
           <item row="10" column="0" colspan="2">
            <widget class="QLabel" name="orderPackagingLabel">
             <property name="text">
              <string>Select the packaging (checks all items above and marks order as packed):</string>
             </property>
            </widget>
           </item>
           <item row="11" column="0" colspan="2">
            <widget class="QComboBox" name="packagingComboBox"/>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QLabel" name="orderIdLabel">
             <property name="font">
              <font>
               <pointsize>10</pointsize>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Order #0</string>
             </property>
             <property name="openExternalLinks">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QPushButton" name="startOverButton">
       <property name="text">
        <string>&amp;Start over</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton2">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../res.qrc"/>
 </resources>
 <connections/>
</ui>
